app/actions/battles.py

```diff
@@ -0,0 +1,8 @@
+def duel(knight1: dict, knight2: dict) -> dict:
+    knight1["hp"] -= (knight2["power"] - knight1["add_some_abilities"])

```

Comment:
There is a potential issue here if 'knight2["power"]' is less than 'knight1["add_some_abilities"]'. This would result in increasing 'knight1["hp"]', which might not be the intended behavior. Consider adding a check to ensure that the subtraction does not result in a negative value.